src/taxpy/deductions/qbi_range.py

```diff
@@ -47,18 +50,38 @@ class QbiRange:
             "m": [364_200, 464_200]
         },
     }
+    
+    def __init__(self, year: int = current_year):
+        """
+
+        :param year: The relevant tax year.
+        """
+        # init at top.
+        self.year = year
+
+        self.s_lower = None
+        self.s_upper = None
+        self.s_phase_in = None
+
+        self.m_lower = None
+        self.m_upper = None
+        self.m_phase_in = None
+
+        self.define_qbi()
 
     def define_qbi(self) -> None:
         """
 
         :return: Nothing.
         """
+        # TODO: Allow the programmer to assign values for years which the class does not by default contain.
```

Comment:
If the class does not contain a year, then there is no QBI deduction for that year. QBI is only for 2018 - 2025 tax years. However, override_qbi() exists if they want to change stuff for some reason.